halo/hooks/object.cpp

```diff
@@ -28,9 +32,240 @@ Signature(true, sig_biped_jump,
     {0x8B, 0x0D, -1, -1, -1, 0x00,
      0x8B, 0x51, 0x34, 0x83, 0xEC, 0x10, 0x53, 0x55}); // size 6 do manual
 
+#define OBJECT_BEHAVIOR_FUNCTION_PTRS(type) \
+    ObjAdjustPlacement            type ## _adjust_placement              = NULL; \
+    ObjCreate                     type ## _create                        = NULL; \
+    ObjPlace                      type ## _place                         = NULL; \
+    ObjDestroy                    type ## _destroy                       = NULL; \
+    ObjUpdate                     type ## _update                        = NULL; \
+    ObjExportFunctionValues       type ## _export_function_values        = NULL; \
+    ObjHandleDeletedObject        type ## _handle_deleted_object         = NULL; \
+    ObjHandleRegionDestroyed      type ## _handle_region_destroyed       = NULL; \
+    ObjHandleParentDestroyed      type ## _handle_parent_destroyed       = NULL; \
+    ObjPreprocessNodeOrientations type ## _preprocess_node_orientations  = NULL; \
+    ObjPostprocessNodeMatricies   type ## _postprocess_node_matricies    = NULL; \
+    ObjReset                      type ## _reset                         = NULL; \
+    ObjDisconnectFromStructureBsp type ## _disconnect_from_structure_bsp = NULL; \
+    ObjNotifyImpulseSound         type ## _notify_impulse_sound          = NULL; \
+    ObjRenderDebug                type ## _render_debug                  = NULL; \
+    ObjNewToNetwork               type ## _new_to_network                = NULL; \
+    ObjUpdateBaseline             type ## _update_baseline               = NULL; \
+    ObjGenerateUpdateHeader       type ## _generate_update_header        = NULL; \
+    ObjHandleReceivedUpdate       type ## _handle_received_update        = NULL; \
+    ObjShouldDoUpdateIncremental  type ## _should_do_update_incremental  = NULL; \
+    ObjSetLastUpdateTime          type ## _set_last_update_time          = NULL; \
+    // End of Macro
+
+ObjectBehaviorDefinition new_obje_beh;
```

Comment:
From our voice conversation --These values are all enumerated out (instead of being an array, for example) for easier access via assembly? If so, that's an important note to put here.